builtin/mainmenu/common.lua

```diff
@@ -300,3 +300,34 @@ function is_server_protocol_compat_or_error(proto_min, proto_max)
 
 	return true
 end
+
+--------------------------------------------------------------------------------
+local function menu_load_worldmt(selected)
```

Comment:
I just read @ShadowNinja's review on #2586, the code redundancy were these lines:

```Lua
 if menudata.worldlist:get_list()[selected] ~= nil then
	local filename = world.path .. DIR_DELIM .. "world.mt"  
	local world_conf = Settings(filename)

	-- specific stuff here
end
```

which yours doesn't fix.